### PR TITLE
fix(map.jinja): Fix useflags regex

### PR DESCRIPTION
**Pull request by mgomersbach**
### PR progress checklist (to be filled in by reviewers)

- [x] Changes to documentation are appropriate (or tick if not required)
- [x] Changes to tests are appropriate (or tick if not required)
- [x] Reviews completed

---

#### Primary type

- [ ] `[build]`    Changes related to the build system
- [ ] `[chore]`    Changes to the build process or auxiliary tools and libraries such as documentation generation
- [ ] `[ci]`       Changes to the continuous integration configuration
- [ ] `[feat]`     A new feature
- [x] `[fix]`      A bug fix
- [ ] `[perf]`     A code change that improves performance
- [ ] `[refactor]` A code change that neither fixes a bug nor adds a feature
- [ ] `[revert]`   A change used to revert a previous commit
- [ ] `[style]`    Changes that do not affect the meaning of the code (white-space, formatting, missing semi-colons, etc.)

#### Secondary type

- [ ] `[docs]`     Documentation changes
- [ ] `[test]`     Adding missing or correcting existing tests

### Does this PR introduce a `BREAKING CHANGE`?

No.

### Describe the changes you're proposing

When using version inside useflag, the sls_block macro with ebuildpkg regex result in a range error.
Replacing the version argument with uses seem to work as escapes don't really work there.

### Pillar / config required to test the proposed changes

Current master branch has non working targets

### Debug log showing how the proposed changes work

```
----------
          ID: php_install_imagick_dev-php-pecl-imagick
    Function: pkg.installed
        Name: dev-php/pecl-imagick
      Result: False
     Comment: An exception occurred in this state: Traceback (most recent call last):
                File "/usr/lib64/python3.6/site-packages/salt/state.py", line 1933, in call
                  **cdata['kwargs'])
                File "/usr/lib64/python3.6/site-packages/salt/loader.py", line 1951, in wrapper
                  return f(*args, **kwargs)
                File "/usr/lib64/python3.6/site-packages/salt/states/pkg.py", line 1563, in installed
                  **kwargs)
                File "/usr/lib64/python3.6/site-packages/salt/states/pkg.py", line 651, in _find_install_targets
                  for name, version in desired.items()
                File "/usr/lib64/python3.6/site-packages/salt/states/pkg.py", line 652, in <listcomp>
                  if not (name in cur_pkgs and (version is None or _fulfills_version_string(cur_pkgs[name], version)))
                File "/usr/lib64/python3.6/site-packages/salt/states/pkg.py", line 216, in _fulfills_version_string
                  fullfills_all = fullfills_all and _fulfills_version_spec([installed_version], operator, version_string, ignore_epoch=ignore_epoch)
                File "/usr/lib64/python3.6/site-packages/salt/states/pkg.py", line 234, in _fulfills_version_spec
                  if (oper == '==' and fnmatch.fnmatch(ver, desired_version)) \
                File "/usr/lib64/python3.6/fnmatch.py", line 36, in fnmatch
                  return fnmatchcase(name, pat)
                File "/usr/lib64/python3.6/fnmatch.py", line 70, in fnmatchcase
                  match = _compile_pattern(pat)
                File "/usr/lib64/python3.6/fnmatch.py", line 46, in _compile_pattern
                  return re.compile(res).match
                File "/usr/lib64/python3.6/re.py", line 233, in compile
                  return _compile(pattern, flags)
                File "/usr/lib64/python3.6/re.py", line 301, in _compile
                  p = sre_compile.compile(pattern, flags)
                File "/usr/lib64/python3.6/sre_compile.py", line 562, in compile
                  p = sre_parse.parse(p, flags)
                File "/usr/lib64/python3.6/sre_parse.py", line 855, in parse
                  p = _parse_sub(source, pattern, flags & SRE_FLAG_VERBOSE, 0)
                File "/usr/lib64/python3.6/sre_parse.py", line 416, in _parse_sub
                  not nested and not items))
                File "/usr/lib64/python3.6/sre_parse.py", line 765, in _parse
                  p = _parse_sub(source, state, sub_verbose, nested + 1)
                File "/usr/lib64/python3.6/sre_parse.py", line 416, in _parse_sub
                  not nested and not items))
                File "/usr/lib64/python3.6/sre_parse.py", line 553, in _parse
                  raise source.error(msg, len(this) + 1 + len(that))
              sre_constants.error: bad character range 7-3 at position 20
     Started: 14:28:26.927898
    Duration: 6321.149 ms
     Changes:   
```

### Documentation checklist

- [ ] Updated the `README` (e.g. `Available states`).
- [ ] Updated `pillar.example`.

### Testing checklist

- [ ] Included in Kitchen (i.e. under `state_top`).
- [ ] Covered by new/existing tests (e.g. InSpec, Serverspec, etc.).
- [ ] Updated the relevant test pillar.

### Additional context

Before it got moved from ng namespace, I created a lot of this mess.

### CODE REVIEW

**Comment by myii** [comment]
Sorry for the delay @mgomersbach.  We don't have a suitable test environment for Gentoo and it doesn't seem easy to achieve, either.

So this continues from #156.  I understand the need for the changes (i.e. `version`) but I'm not quite sure what the changes achieve.

* Of course, it shouldn't be referring to `version` but what exactly is `uses`?
* Is that supposed to be the [`use`](https://docs.saltstack.com/en/latest/ref/states/requisites.html#use) requisite?
* And if so, what state is referred to by `'[php_targets_php' + useversion + ']'`?

File: php/map.jinja [hunk 1]
```diff
@@ -2820,57 +2820,68 @@
             },
             'Gentoo' : {
                 'pkgs': {
+                    'adodb': 'dev-php/adodb',
                     'apache2': [{'name': 'dev-lang/php','version': ':' + php_version + '[apache2]'}],
-                    'apc': [{'name': 'dev-php/pecl-apcu_bc','version': '[php_targets_php' + useversion + ']'}],
-                    'apcu': [{'name': 'dev-php/pecl-apcu','version': '[php_targets_php' + useversion + ']'}],
+                    'apc': [{'name': 'dev-php/pecl-apcu_bc','uses': '[php_targets_php' + useversion + ']'}],
+                    'apcu': [{'name': 'dev-php/pecl-apcu', 'uses': "[php_targets_php" + useversion + "]"}],
```
Comment (myii) [inline php/map.jinja]: Can we stick to single-quoting to maintain consistency?  Meaning:

```suggestion
                    'apcu': [{'name': 'dev-php/pecl-apcu', 'uses': '[php_targets_php' + useversion + ']'}],
```

**Comment by mgomersbach** [comment]
> Sorry for the delay @mgomersbach. We don't have a suitable test environment for Gentoo and it doesn't seem easy to achieve, either.

I know, Gentoo is not high profile, so any support at all is greatly appreciated.

> 
> So this continues from #156. I understand the need for the changes (i.e. `version`) but I'm not quite sure what the changes achieve.
> 
>     * Of course, it shouldn't be referring to `version` but what exactly is `uses`?

"uses" is from the [ebuildpkg](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.ebuildpkg.html) module.

Most features listed here come from compiles the package "dev-lang/php" with a useflag. And external ones have useflags with versions in it.
A dash between numbers in the useflag (like: [php_targets_php7-3] ) generates a error like in the error report. 
Somewhere it gets interpreted as regex syntax, and was not able to generate the right escape sequence around it.

**Comment by myii** [comment]
Merged, thanks for the contribution @mgomersbach.

> I know, Gentoo is not high profile, so any support at all is greatly appreciated.

Not a problem, this PR has stimulated some discussion about how to get tests running on Gentoo.  It would be great if you could bring any of your suggestions, such as how to install a specific version of Salt:

* https://github.com/netmanagers/salt-image-builder/issues/19

> "uses" is from the [ebuildpkg](https://docs.saltstack.com/en/latest/ref/modules/all/salt.modules.ebuildpkg.html) module.

Great, that's exactly what I needed to know.  At least we have some reference to that now in this discussion.

**Pull request merged**